### PR TITLE
fix(helm): headless service FQDN define.

### PR DESCRIPTION
**Pull request by kii-la**

### CODE REVIEW

**Review by zitsen — CHANGES_REQUESTED** [review]
We use angular convential changelog as commit message style - for changelog auto generation, in this project :D.

See https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular

File: helm/tdengine/templates/statefulset.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ spec:
           - name: SERVICE_NAME
             value: {{ include "tdengine.fullname" . }}
           - name: STS_NAME
-            value: {{ .Chart.Name }}
+            value: {{ .Release.Name }}
```
[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/statefulset.yaml:51]: if `STS_NAME: {{ include "tdengine.fullname" . }}` 

```yaml
          - name: TAOS_FIRST_EP
            value: "$(STS_NAME)-0.$(SERVICE_NAME).$(STS_NAMESPACE).svc.cluster.local:$(TAOS_SERVER_PORT)"
```

Is it correct to use `STS_NAME`  in this place?

File: helm/tdengine/templates/service.yaml [hunk 2]
```diff
@@ -18,3 +18,25 @@ spec:
     {{- end }}
   selector:
     {{- include "tdengine.selectorLabels" . | nindent 4 }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "tdengine.fullname" . }}-headless
```
[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/service.yaml:25]: Usually, the default service is the Loadbalancer, and headless is the declaration that points to the corresponding Pod for statefulsets.

[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/service.yaml:25]: and after testing, the headless service FQDN does not work without an additional service.

File: helm/tdengine/templates/statefulset.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zitsen) [reply helm/tdengine/templates/statefulset.yaml:51]: Test it with `helm install ` with a different release name such as `taosd`

File: helm/tdengine/templates/service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zitsen) [reply helm/tdengine/templates/service.yaml:25]: The helm package is tested multiple times in rke2/minikube, I don't know what the `FQDN does not work` meaning, could you give me an test pipeline ?

File: helm/tdengine/templates/statefulset.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/statefulset.yaml:51]: Yes, you're right. 
`STS_NAME` means the name of the stateful set, which should match the field in the stateful set metadata.

[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/statefulset.yaml]: Ok, I'm trying to finish preparing the new test environment, which was previously having the problem with a cluster built on RKE. I'm going to re-implement it here using kubeadmin.

File: helm/tdengine/templates/service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kii-la) [reply helm/tdengine/templates/service.yaml:25]: Ok, I'm trying to finish preparing the new test environment, which was previously having the problem with a cluster built on RKE. I'm going to re-implement it here using kubeadmin.

**Comment by zitsen** [comment]
If you test it ok, could you re-request review?

**Pull request closed without merging**